### PR TITLE
Olelensmar/docs/ongoing updates

### PR DESCRIPTION
**Pull request by olensmar**
ongoing documentation updates...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,9 +27,14 @@ Main Kubtest components are:
 
 - kubectl plugin - simple - installed w/o 3rd party repositories (like Krew etc), communicates with
 - API Server - work orchestrator, runs executors, gather execution results
-- CRDs Operator - watch Kubtest CR, handles changes communicates with API Server
+- [CRDs Operator](https://github.com/kubeshop/kubtest-operator) - watch Kubtest CR, handles changes communicates with API Server
```
Comment (jasmingacic) [inline README.md:30]: watch -> watches

**Review by exu — APPROVED** [review]
We could also add link to kubectl kubtest plugin

**Review by nicufk — APPROVED** [review]

**Comment by exu** [comment]
@olensmar merge main branch as this one fails

File: docs/custom-executor.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+In this example for sake of simplicity we'll use `kubtest` components to implement executor they help us with:
+- Defining basic HTTP Rest based server skeleton
+- Creating job queue - our tests will be run in async mode
+- Running runners - this is the only part which need to be implmented when using `kubtest` components.
+
+
+To implement new executor we should do following thigs: 
+
+- Define an input format for the tests.
+  In order to communicate effectivele with executor we need to define a format on how to structure the tests. And bellow is an example for the curl based tests.
```
Comment (jasmingacic) [inline docs/custom-executor.md:26]:
```suggestion
  In order to communicate effectively with executor we need to define a format on how to structure the tests. And bellow is an example for the curl based tests.
```

File: docs/custom-executor.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+            memory: 200Mi
+          requests:
+            cpu: 100m
+            ephemeral-storage: 1Gi
+            memory: 100Mi
+```
+
+(You should tune deployment for production use)
+
+Last thing which need to be done is to create binding for your executor to some type
```
Comment (jasmingacic) [inline docs/custom-executor.md:210]:
```suggestion
Last thing which needs to be done is to create binding for your executor to some type 
```

File: docs/reference.md [hunk 4]
```diff
@@ -0,0 +1,327 @@
+# Reference for 'kubectl kubtest` plugin
```
Comment (jasmingacic) [inline docs/reference.md:1]: We can point to cobra generated CLI reference here -> https://github.com/kubeshop/kubtest/tree/main/docs/cli

Comment (olensmar) [reply docs/reference.md:1]: added links to these to the menu

**Review by jasmingacic — APPROVED** [review]

**Pull request merged**